### ISSUE TITLE
[Bar chart] TapTool callback called multiple times

### ISSUE THREAD

**Issue by wsfax** [bokeh/bokeh#2976]
When I click on a rectangle I get the message "pressed" 3 times !!
I was expecting to see it only once.

Anyone has an explanation for this behavior?

```
from bokeh.charts import Bar, show
from bokeh.models import TapTool, CustomJS

data = {'ID': [0, 1, 2], 'Total': [5.0, 20.0, 1.0]}

p = Bar(data, label='ID', values='Total', width=400, tools='tap')

tap = p.select(dict(type=TapTool))
tap.callback = CustomJS(code='''alert("pressed");''')

show(p)
```

**Comment by bryevdv** [comment]
The reason is because by default the hover tool attaches itself to all `GlyphRenderer` objects that are present. In this case there are three. Not sure offhand why, @rothnic do you have input?

**Comment by wsfax** [comment]
@bryevdv but I hope that you agree with me that this is not what someone would like to see

**Comment by bryevdv** [comment]
I was certainly not disagreeing with you, merely offering information about the technical proximate cause to move the discussion along.

**Comment by wsfax** [comment]
thanks for your reply.
I actually got confused because this issue was labeled as "discussion" and not "bug". At least from a noob point of view it looks like a bug.

**Comment by bryevdv** [comment]
It could be a `bug` or a `feature`, more investigation is still needed. If the situation with multiple renderers is intentional and correct, then we need to add conveniences that make correct usage of tools with that context easier (a feature). If the multiple renderers is an error, then we need to correct it (a bug).

**Comment by rothnic** [comment]
@fpliger and @bryevdv, We had a discussion early on about where the data sources live, etc. We decided to make more of them compared using namespaces to separate things. 

At the moment, this chart produces 3 glyph renderers because it represents 3 unique "groups" of data. I think we talked about using tags or something along those lines to provide identifiers once the python-only models are materialized into glyph renderers on the javascript side. I'm not familiar with the exact machinery for customjs, but It seems like on press, you'd have some call back where you'd check some value to know what was pressed.
